segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/LogReader.java

```diff
@@ -141,11 +144,12 @@ private void openNextLedger(LedgerAddress address) throws DurableDataLogExceptio
 
         ReadLedger previousLedger;
         try {
-            val reader = Exceptions.handleInterruptedCall(
-                    () -> ledger.readEntries(address.getEntryId(), lastEntryId));
+            LedgerEntries entries = Exceptions.handleInterruptedCall(
+                    () -> ledger.read(address.getEntryId(), lastEntryId));
             previousLedger = this.currentLedger;
-            this.currentLedger = new ReadLedger(metadata, ledger, reader);
+            this.currentLedger = new ReadLedger(metadata, ledger, entries);
             if (previousLedger != null) {
+                previousLedger.release();
```

Comment:
Here you are invoking `release` before `Ledgers.close`, but everywhere else you did it after. Is this a problem?

Reply:
It shouldn't be a problem, as *release* is only about releasing Netty memory.
I will move this line after *close* in order to have a consistent pattern.